### ISSUE TITLE
Method to estimate the amount to compensate

### ISSUE THREAD

**Issue by marceljay** [IsEthereumGreen/contract#1]
First of all, thanks for this fantastic idea. I have some questions regarding the general logic:

1) > `sumCO2TokensBurned` starts at 0 and can be increased by sending funds to the contract. 

What assets do you imagine being sent to the contract? Regular tokens like DAI, other stables, ETH?
    Because I think it should be possible to send accepted/whitelisted carbon tokens as well, which then can be automatically offset by the contract. 
    
 2) > `costToCompensateUntilNow`
 
 Should really be a public view function outputting a range, or at least stating the assumptions. The costs are not equal the emissions to be compensated and can vary tremendously depending on the quality of the credits/offsets.
 
 This means we first need to come up with a valid methodology on how these emissions are being calculated, e.g. an average value that is multiplied by block numbers since a certain date. Tricky because hashpower and electricity sources can change all the time.

**Comment by timschloe** [comment]
In terms of climate justice, it's always better to be conservative than precise. Just assume 100% Diesel-generators for all electricity and compensate that. 
Then you're on the safe side and helped a little to move towards a sustainable future instead of just being able to say "we compensated exactly our footprint and not a ton more".

**Comment by johnx25bd** [comment]
+100 on your comments Tim, estimation methodologies are really divergent and "net zero" isn't good enough. 

The [methodology](https://www.notion.so/offsetra/Carbon-FYI-Methodology-51e2d8c41d1c4963970a143b8629f5f9) used at [carbon.fyi](https://carbon.fyi) is quite a robust starting point.

**Comment by mjohnson518** [comment]
I agree the carbon.fyi methodology is robust, but it still relies on generalized, annualized grid emissions factors (as do almost all carbon accounting methodologies for all industries). 

The problem with using generalized, annualized grid emissions factors is that they are 1 static number that represents an average of an entire country or regions energy grid emissions intensity, and they do not take into account the energy grid dynamics of specific locations. 

What this means is that methodologies do not take into account the different energy grid mixes across countries or regions (i.e. there is no recognition of an abundance of wind resources in West Texas vs an abundance of Coal plants in West Pennsylvania). 

In the world of Carbon Accounting, there are three different methods for this form of emissions accounting: consequential, market-based attributional, and location-based attributional. 

The consequential approach uses **_marginal emissions factors_**, which are based on the incremental changes in GHG emissions from electricity grids due to changes in demand. The market-based method uses emission factors based on the specific electricity suppliers and sources that the miner has chosen to use, as well as any energy attribute certificates (such as renewable energy certificates, or RECs) that they have purchased. The location-based method uses the average emission factors of the electricity grids from which consumption occurs.

The consequential method looks at **_the actual additional or marginal impact that miners create by choosing to draw energy from a specific grid_**. It avoids arbitrarily sharing responsibility for all emissions among all grid users.  It shows the macro impact that the electricity consumption has on the local grid. For example, even if a miner enters a clean energy contract, by using up some of the locally available clean energy they may indirectly push a different electricity consumer to use a high-emission energy source. 

I've been coordinating some discussions on this topic with devs & researchers on the [Merge 4 Climate Discord](https://discord.gg/ka59kR6u) that Rapha, Alex Taylor & Ligi made last year. 

Come join the party and lets get this convo moving full steam ahead!

Relevant links: 
- https://cryptoclimate.org/solutions/
- https://www.marcjohnson.info/CCA.html
- https://kylemcdonald.github.io/ethereum-emissions/

**Comment by ligi** [comment]
I agree getting these numbers is one of the most tricky parts with the endeavor - but I also agree the perfect is the enemy of the good here - so I think good educated guesstimates (e.g. with the carbon.fyi) method with some safety margin to get to a upper bound and a good chance to compensate (more than) enough is a OK approach for the MVP so we can get started without being blocked by bike-shedding. But we sure need an upgrade path to improve here in the future.

**Comment by ligi** [comment]
Via chat we planned a call to discuss issues like this: https://github.com/IsEthereumGreen/org/blob/main/call0.ics
